rtloader/common/builtins/datadog_agent.c

```diff
@@ -411,6 +419,41 @@ static PyObject *log_message(PyObject *self, PyObject *args)
     Py_RETURN_NONE;
 }
 
+/*! \fn PyObject *send_log(PyObject *self, PyObject *args)
+    \brief This function implements the `datadog_agent.send_log` method, sending
+    a log for eventual submission.
+    \param self A PyObject* pointer to the `datadog_agent` module.
+    \param args A PyObject* pointer to a 2-ary tuple containing a log line and the
+    unique ID of a check instance.
+    \return A PyObject* pointer to `None`.
+
+    This function is callable as the `datadog_agent.send_log` Python method and
+    uses the `cb_send_log()` callback to retrieve the value from the agent
+    with CGO. If the callback has not been set `None` will be returned.
+*/
+static PyObject *send_log(PyObject *self, PyObject *args)
+{
+    // callback must be set
+    if (cb_send_log == NULL) {
+        Py_RETURN_NONE;
+    }
+
+    char *log_line, *check_id;
```

Comment:
If I understand Python's documentation correctly, these should be `const char*` as the memory is owned by python

This would imply the `cb_send_log_t` type should also be updated to pass both parameters as pointers to constant chars

Reply:
This is the same as every function in this file; are you certain?

Reply:
Quite certain:
> Other formats take a str or a read-only bytes-like object, such as bytes, and provide a const char * pointer to its buffer. In this case the buffer is “borrowed”: it is managed by the corresponding Python object, and shares the lifetime of this object. You won’t have to release any memory yourself.

https://docs.python.org/3/c-api/arg.html#strings-and-buffers

It doesn't make an actual difference for most purposes, but having a `const char'` makes it clear that we don't own that memory

Reply:
Thanks! This will be done in a follow-up PR.